### PR TITLE
DEV: Pass messageId as a dynamic segment instead of a query param

### PR DESCRIPTION
**Pull request by romanrizzi**
Depends on https://github.com/discourse/discourse/pull/19782

### CODE REVIEW

**Comment by romanrizzi** [comment]
@jjaffeux - In theory, this should be ready for review. I pushed one more commit to fix some issues with drawer navigations. There are two tests that are consistently failing, but seem unrelated to the changes this PR introduces.

Will investigage a bit more tomorrow.

File: plugins/chat/assets/javascripts/discourse/components/reviewable-chat-message.hbs [hunk 1]
```diff
@@ -1,8 +1,11 @@
 <div class="flagged-post-header">
   <LinkTo
     @route="chat.channel"
-    @models={{this.chatChannel.routeModels}}
-    @query={{hash messageId=@reviewable.target_id}}
+    @models={{array
+      this.chatChannel.slugifiedTitle
+      this.chatChannel.id
+      @reviewable.target_id
+    }}
```
Comment (jjaffeux) [inline plugins/chat/assets/javascripts/discourse/components/reviewable-chat-message.hbs:8]: routeModels?

File: plugins/chat/assets/javascripts/discourse/routes/chat-channel.js [hunk 2]
```diff
@@ -12,27 +10,27 @@ export default class ChatChannelRoute extends DiscourseRoute {
...
 
-  @action
-  didTransition() {
-    const { channelId, messageId } = this.paramsFor(this.routeName);
-    if (channelId && messageId) {
-      schedule("afterRender", () => {
-        this.chat.openChannelAtMessage(channelId, messageId);
-        this.controller.set("messageId", null); // clear the query param
-      });
+    // Rewrite URL with slug in the child route to preseve
```
Comment (jjaffeux) [inline plugins/chat/assets/javascripts/discourse/routes/chat-channel.js:28]:
```suggestion
    // Rewrite URL with slug in the child route to preserve
```

**Comment by jjaffeux** [comment]
Im a little bit concerned by the code duplication here, did you try to implement something like this?

```javascript
    this.route(
      "channel",
      { path: "/c/:channelTitle/:channelId/*messageId" },
      function () {
        this.route("info", { path: "/info" }, function () {
          this.route("about", { path: "/about" });
          this.route("members", { path: "/members" });
          this.route("settings", { path: "/settings" });
        });
      }
    );
```

**Review by jjaffeux — APPROVED** [review]

**Pull request merged**